assets/blocks/quiz/data.js

```diff
@@ -143,13 +167,19 @@ export function createQuestionBlock( question ) {
 /**
  * Find a question block based on question ID, or title if ID is missing.
  *
- * @param {Array}        blocks
- * @param {QuizQuestion} item
+ * @param {Array}                             blocks
+ * @param {QuizQuestion|QuizCategoryQuestion} item
  */
-export const findQuestionBlock = ( blocks, { id, title } ) => {
+export const findQuestionBlock = (
+	blocks,
+	{ id, title, options: { category } }
+) => {
 	const compare = ( { attributes } ) =>
 		id === attributes.id ||
-		( ! attributes.id && attributes.title && attributes.title === title );
+		( ! attributes.id && attributes.title && attributes.title === title ) ||
+		( ! attributes.id &&
+			attributes.options.category &&
```

Comment:
This could use an extra null check on `options`

Reply:
028ebff478c657d20fabddfe464609c2a071568d